datasets/_global/gem_energy_ownership/crawler.py

```diff
@@ -96,16 +90,19 @@ def crawl_company(context: Context, row: Dict[str, str], skipped: Set[str]):
     if homepage:
         entity.add("website", split_urls(homepage))
     if schema != "Person":
-        entity.add("permId", perm_id)
-        # find a way to remap invalid ones
-        entity.add("ogrnCode", ru_id)
+        entity.add_cast("Company", "permId", perm_id)
+        # Check if ru_id is a valid OGRN
+        if ru_id and rigour.ids.OGRN.is_valid(ru_id):
+            entity.add("ogrnCode", ru_id)
+        else:  # Remap invalid ones
+            entity.add("registrationNumber", ru_id)
```

Comment:
sorry for changing my mind at the last minute. This was about 20 items, right? If it's more like 100, maybe this is good, but for 20ish I think we should use the lookups mechanism instead of adding another special case to the crawler. We can kinda do it in bulk, right? like

```yaml
    - match:
        - id1
        - id2
        - ...
      prop: registrationNumber
```